### PR TITLE
refactor(ngx-conf) better lua_shared_dict citizens

### PR DESCRIPTION
**Pull request by thibaultcha**
### Full changelog

* prefix our shared dicts with `kong_*`
* do not unnecessarily create the C* shared dict
* bump hard-coded size values

### CODE REVIEW

File: kong/templates/nginx_kong.lua [hunk 1]
```diff
@@ -29,11 +29,13 @@ lua_code_cache ${{LUA_CODE_CACHE}};
 lua_socket_pool_size ${{LUA_SOCKET_POOL_SIZE}};
 lua_max_running_timers 4096;
 lua_max_pending_timers 16384;
-lua_shared_dict kong 4m;
-lua_shared_dict cache ${{MEM_CACHE_SIZE}};
-lua_shared_dict process_events 1m;
-lua_shared_dict cluster_events 1m;
-lua_shared_dict cassandra 5m;
+lua_shared_dict kong                4m;
```
Comment (thibaultcha) [inline kong/templates/nginx_kong.lua]: Not creative enough to find a good name for this one. It really is pure Kong stuff, non-perishable an supposed to be long-lived (although I do think we put some locks in there, to be audited).

Comment (thefosk) [inline kong/templates/nginx_kong.lua]: All the locks should be in the `cache_locks`, and if they are not, that would be a mistake

Comment (thibaultcha) [inline kong/templates/nginx_kong.lua]: Not necessarily. As long as they are not in `kong`.

File: kong/templates/nginx_kong.lua [hunk 2]
```diff
@@ -29,11 +29,13 @@ lua_code_cache ${{LUA_CODE_CACHE}};
...
+lua_shared_dict kong_cache          ${{MEM_CACHE_SIZE}};
+lua_shared_dict kong_process_events 1m;
+lua_shared_dict kong_cluster_events 1m;
+> if database == "cassandra" then
+lua_shared_dict kong_cassandra      5m;
+> end
```
Comment (Tieske) [inline kong/templates/nginx_kong.lua]: On mobile, but this doesn't seem to reflect the last commit where all sizes were set to 5 m?

That said, any rationale to the 5m sizes?

Comment (thibaultcha) [inline kong/templates/nginx_kong.lua]: Taken care of. No rationale right now, except that 5 > 1 and thus safer, while still in the same order of magnitude.

**Pull request merged**